Doc adjustments (#169)
cscherrer/MeasureTheory.jl: docs/src/affine.md

It's very common for measures to use parameters `μ` and `σ`, for example as in `Normal(μ=3, σ=4)` or `StudentT(ν=1, μ=3, σ=4)`. In this context, `μ` and `σ` need not always refer to the mean and standard deviation (the `StudentT` measure specified above is equivalent to a [Cauchy](https://en.wikipedia.org/wiki/Cauchy_distribution) measure, so both mean and standard deviation are undefined).

In general, `μ` is a "location parameter", and `σ` is a "scale parameter". Together these parameters determine an affine transformation.

```math
f(z) = σ z + μ
```

Starting with the above definition, we'll use ``z`` to represent an "un-transformed" variable, typically coming from a measure which has neither a location nor a scale parameter, for example `Normal()`.

It is a common misconception that affine transformations are "linear". Linearity requires ``f(ax + by) = a f(x) + b f(y)`` for scalars ``a`` and ``b``, which would only holds for the above affine transformation ``f`` if ``μ=0``.

## Cholesky-based parameterizations

If the "un-transformed" `z` is univariate, things are relatively simple. But it's important our approach handle the multivariate case as well.

> In the literature, it's common for a multivariate normal distribution to be parameterized by a mean `μ` and covariance matrix `Σ`. This is mathematically convenient, but less idea for efficient computation.

Reviewer: It seems like the point here is that `awkward` is vague. I agree with that, how about
```suggestion
In the literature, it's common for a multivariate normal distribution to be parameterized by a mean `μ` and covariance matrix `Σ`. This is mathematically convenient, but leads an ``O(n^3)`` [Cholesky decomposition](https://en.wikipedia.org/wiki/Cholesky_decomposition), which can be a significant bottleneck.
```

While MeasureTheory.jl includes (or will include) a parameterization using `Σ`, we prefer to work in terms of its Cholesky decomposition ``σ``.

> The relationship between the computationally efficient "``σ``" and more familiar parameteriation `Σ` can be seen as follows:
> 
> Let ``σ`` be a lower-triangular matrix satisfying

Reviewer: I think I see where you're going with this, and I like it. But maybe we can keep the "relationship" perspective and also the conciseness with 
```suggestion
To see the relationship between our ``σ`` parameterization and the likely more familiar  ``Σ`` parameterization,  let ``σ`` be a lower-triangular matrix satisfying
```

# Affine Transformations

It's very common for measures to use parameters `μ` and `σ`, for example as in `Normal(μ=3, σ=4)` or `StudentT(ν=1, μ=3, σ=4)`. In this context, `μ` and `σ` need not always refer to the mean and standard deviation (the `StudentT` measure specified above is equivalent to a [Cauchy](https://en.wikipedia.org/wiki/Cauchy_distribution) measure, so both mean and standard deviation are undefined).

In general, `μ` is a "location parameter", and `σ` is a "scale parameter". Together these parameters determine an affine transformation.

```math
f(z) = σ z + μ
```

Starting with the above definition, we'll use ``z`` to represent an "un-transformed" variable, typically coming from a measure which has neither a location nor a scale parameter, for example `Normal()`.

> Affine transformations are often mistakenly referred to as "linear". In fact, an affine transformation is ["the composition of two functions: a translation and a linear map"](https://en.wikipedia.org/wiki/Affine_transformation#Representation). For a function `f` to be linear requires 

Reviewer 2: ```suggestion
Affine transformations are often ambiguously referred as "linear transformations". In fact, an affine transformation is ["the composition of two functions: a translation and a linear map"](https://en.wikipedia.org/wiki/Affine_transformation#Representation) in the stricter algebraic sense: For a function `f` to be linear requires 
```

Author: Thanks!

Reviewer: Looks like you may have accepted just as Moritz was suggesting an alternative. I think I like his suggestion better than mine, though I'd keep the "as" (maybe a US vs European difference?)

Author: Think it’s now incorporated. :)

> Affine transformations are often ambiguously referred as "linear transformations". In fact, an affine transformation is ["the composition of two functions: a translation and a linear map"](https://en.wikipedia.org/wiki/Affine_transformation#Representation) in the stricter algebraic sense: For a function `f` to be linear requires 

Reviewer 3: Is this paragraph really useful? I would assume most of our readers know the difference between linear and affine

Reviewer 2: Yes, but we should say what *affine* is anyway. But maybe just a brief definition and a link.

*An [affine function](https://en.wikipedia.org/wiki/Affine_transformation) `f(x) = A*x + b`  is just a linear map `A` translated by a constant `b` with constant `b` and linear map/operator `A` .

Reviewer: I think those knowing the difference are limited to a small neighborhood around "math PhD programs". We need to be very clear that there's a difference (otherwise many will think we're just using `Affine` to sound fancy). But the original might be too pedantic about it. Having something more brief is fine, and maybe preferable.

``f(ax + by) == a f(x) + b f(y)``
for scalars ``a`` and ``b``. For an affine function
``f(z) = σ * z + μ``, where the linear map is defined as ``σ`` and the translation defined as ``μ``,
linearity holds only if the translation component ``μ`` is equal to zero.

## Cholesky-based parameterizations

If the "un-transformed" `z` is univariate, things are relatively simple. But it's important our approach handle the multivariate case as well.

> In the literature, it's common for a multivariate normal distribution to be parameterized by a mean `μ` and covariance matrix `Σ`. This is mathematically convenient, but leads an ``O(n^3)`` [Cholesky decomposition](https://en.wikipedia.org/wiki/Cholesky_decomposition), which becomes a significant bottleneck to compute as ``n`` gets large.

Reviewer 3: leads to

Reviewer 3: ```suggestion
In the literature, it's common for a multivariate normal distribution to be parameterized by a mean `μ` and covariance matrix `Σ`. This is mathematically convenient, but leads to an ``O(n^3)`` [Cholesky decomposition](https://en.wikipedia.org/wiki/Cholesky_decomposition), which becomes a significant bottleneck to compute as ``n`` gets large.
```

While MeasureTheory.jl includes (or will include) a parameterization using `Σ`, we prefer to work in terms of its Cholesky decomposition ``σ``.

To see the relationship between our ``σ`` parameterization and the likely more familiar  ``Σ`` parameterization,  let ``σ`` be a lower-triangular matrix satisfying

```math

> σ σᵗ = Σ

Reviewer 3: I don't like this notation for the transpose, I very much prefer $\sigma^\top$, is it the Frenchman in me?

Reviewer: I always found "⊤ is $\top$, and that starts with a T, and that stands for transpose" kind of funny. But I'm open to changing it to anything widely used ($\top$ included), as long as we're consistent

Reviewer 3: Alternately, $\sigma^T$ would be fine. Anything but the small t works for me ^^

Reviewer: This is a simple search and replace (superscripts make it convenient in this case). But maybe a separate PR would be easier? Otherwise it could be easy to miss some cases as we're discussing other changes

Author: Noted here: https://github.com/cscherrer/MeasureTheory.jl/issues/173

# Affine Transformations

It's very common for measures to use parameters `μ` and `σ`, for example as in `Normal(μ=3, σ=4)` or `StudentT(ν=1, μ=3, σ=4)`. In this context, `μ` and `σ` need not always refer to the mean and standard deviation (the `StudentT` measure specified above is equivalent to a [Cauchy](https://en.wikipedia.org/wiki/Cauchy_distribution) measure, so both mean and standard deviation are undefined).

In general, `μ` is a "location parameter", and `σ` is a "scale parameter". Together these parameters determine an affine transformation.

```math
f(z) = σ z + μ
```

Starting with the above definition, we'll use ``z`` to represent an "un-transformed" variable, typically coming from a measure which has neither a location nor a scale parameter, for example `Normal()`.

> Affine transforms are often mistakenly referred to as "linear". For a function `f` to be linear requires 

Author: ```suggestion
Affine transformations are often mistakenly referred to as "linear". In fact, an affine transformation is ["the composition of two functions: a translation and a linear map"](https://en.wikipedia.org/wiki/Affine_transformation#Representation). For a function `f` to be linear requires 
```

``f(ax + by) == a f(x) + b f(y)``
for scalars ``a`` and ``b``. For an affine function

> ``f(z) = σ * z + μ``,

Author: ```suggestion
``f(z) = σ * z + μ``, where the linear map is defined as `σ` and the translation defined as `μ`,
```

> linearity holds only if ``μ = 0``.

Author: ```suggestion
linearity holds only if the translation component ``μ`` is equal to zero.
```

## Cholesky-based parameterizations

If the "un-transformed" `z` is univariate, things are relatively simple. But it's important our approach handle the multivariate case as well.

> In the literature, it's common for a multivariate normal distribution to be parameterized by a mean `μ` and covariance matrix `Σ`. This is mathematically convenient, but leads an ``O(n^3)`` [Cholesky decomposition](https://en.wikipedia.org/wiki/Cholesky_decomposition), which can be a significant bottleneck.

Author: ```suggestion
In the literature, it's common for a multivariate normal distribution to be parameterized by a mean `μ` and covariance matrix `Σ`. This is mathematically convenient, but leads an ``O(n^3)`` [Cholesky decomposition](https://en.wikipedia.org/wiki/Cholesky_decomposition), which becomes a significant bottleneck to compute as ``n`` gets large.
```

𝕍[σ z + μ] = Σ
```

The one-dimensional case where we have

```math
𝕍[σ z + μ] = σ²
```

shows that the lower Cholesky factor of the covariance generalizes the concept of standard deviation, completing the link between ``σ`` and `Σ`.

## The "Cholesky precision" parameterization

The ``(μ,σ)`` parameterization is especially convenient for random sampling. Any measure `z ~ Normal()` determines an `x ~ Normal(μ,σ)` through the affine transformation

```math
x = σ z + μ
```

The log-density transformation of a `Normal` with parameters μ, σ does not follow as directly. Starting with an ``x``, we need to find ``z`` using

```math

> z = σ⁻¹ (x - μ)

Reviewer 3: I think there is a problem with the exponent in the maths formula, why not use $\sigma^{-1}$?

Reviewer: One benefit of having it as unicode when possible is that it can render in the REPL. But we should check that the docs also render well (do they not?)

Reviewer 3: I think they're ok
